src/Avalonia.Ide.CompletionEngine/AssemblyMetadata/MetadataConverter.cs

```diff
@@ -141,6 +149,27 @@ public static Metadata ConvertMetadata(IMetadataReaderSession provider)
                                     true, false, true, true));
                             }
                         }
+
+                        foreach (var fieldDef in typeDef.Fields)
+                        {
+                            if (fieldDef.IsStatic && fieldDef.IsPublic
+                                && 
+                                (fieldDef.ReturnTypeFullName.StartsWith("RoutedEvent") || 
```

Comment:
We should probably check the actual inheritance path here

Reply:
There are 2 problems:
1. As I understand `MedataConverter` abstraction there is no way to pass type to consumers, so we need to determine if field is `RoutedEvent` for all fields. Currently only events use fields, so it may work for now. Other way is to expose type querying api (`ToTypeDefOrRef`) somehow. I think adding `IsRoutedEvent` property is acceptable.

2. We have no reference to `RoutedEvent` to check what's its full name etc. My proposition is to check name and namespace, and add test to guard that `IsRoutedEvent` proeprty is set. Test would reference Avalonia and see if `RoutedEvent` is resolved correctly.

So it would work like this:
```
            bool isRoutedEvent = false;
            ITypeDefOrRef t = f.FieldType.ToTypeDefOrRef();
            while(t != null)
            {
                if(t.Name == "RoutedEvent" && t.Namespace == "Avalonia.Interactivity")
                {
                    isRoutedEvent = true;
                    break;
                }
                t = t.GetBaseType();
            }

            IsRoutedEvent = isRoutedEvent;
```
What you think?

Reply:
Resolving the "routed event" indicator on the metadata reading layer should work, yes.

Reply:
Check new commit